### DISCUSSION TITLE
ATB/Trance larger bars

### DISCUSSION THREAD

**Discussion by Aavock** [Albeoris/Memoria#731] [Ideas]
Hey, very first message here.

The idea is to make the ATB and Trance bars larger to match the PSX version. Is it possible to vertically "stretch" the display zone of both bars and then correct the texture? -I can take care of the texture editing if needed-

Here's a quick preview of what I mean:
![Untitled1](https://github.com/user-attachments/assets/e5c3ea72-4bfa-410b-8611-9526c4bcc5f6)

Thanks to all modders for the outstanding work done so far.

**Comment by snouz** [comment]
Hey Aavock I remember you, you helped me with some font stuff 4y ago.

Yes Indeed, the AT is kinda badly dimensioned. It's probably not very hard to stretch.

**Reply by Aavock** [reply]
Salut snouz! That's correct, glad your remember!

That'd be great since those bars are quite thin. I'll be pleased to edit the texture thereafter.

**Comment by barkermn01** [comment]
This should be possible via the GOProgressBar class, and modifying the RectTransform component of the UIProgressBar,

```c#
            RectTransform rectTransform = ProgressBar.GetComponent<RectTransform>();
            if (rectTransform != null)
            {
                // Set the desired height
                float newHeight = 50f;
                // Get the current size
                Vector2 size = rectTransform.sizeDelta;
                // Modify the height
                size.y = newHeight;
                // Apply the new size
                rectTransform.sizeDelta = size;
            }
```

However i have not implemented that as it raises a question about how we should handle this, should there just be an option for ATB BarSize such as `small, medium, large`, should it be a slider multiplier is size to a limit, or should we just make them bigger always, this requires a convo so should probably be moved to an issue as a Feature Request @snouz would you agree?

On the sizes, 

Small the progress bar is the size it is by default,
Medium the progress bar is half the line height as sugested by OP,
Large the progress bar is the full line height

**Reply by Aavock** [reply]
Thanks barkermn01, I'm not familiar with coding so can you please provide a quick tutorial or -if faster- the modified file (with the values you consider a good balance) so I can start tweaking with the texture?

**Reply by barkermn01** [reply]
This more hear for @snouz and @Tirlititi to come back on it's just an implementation that should work but the developers need to talk about it.

**Reply by Aavock** [reply]
Thanks! Will wait.
